src/components/TasksList/__tests__/TasksList.test.tsx

```diff
@@ -9,11 +14,115 @@ jest.mock('@material-ui/core/Tab', () => 'Tab');
 jest.mock('components/TabPanel', () => 'TabPanel');
 jest.mock('components/AvatarHeader', () => 'AvatarHeader');
 jest.mock('components/NavBar', () => 'NavBar');
+jest.mock('components/TaskListPlaceHolder', () => 'TaskListPlaceHolder');
+jest.mock('components/SnackBar', () => 'SnackBar');
+jest.mock('components/TaskTabsContent', () => 'TaskTabsContent');
 
 describe('Component - TasksList', () => {
-  test('renders correctly', () => {
-    const component = create(<TasksList />);
+  const path = '/buddy/newbies/1234/tasks';
 
-    expect(component.toJSON()).toMatchSnapshot();
+  describe('When there are tasks', () => {
+    const taskListWithDataResponse = [
+      {
+        request: {
+          query: TASK_LIST,
+          variables: {
+            newbieId: '1234',
+          },
+        },
+        result: {
+          data: {
+            newbie: {
+              buddyTasks: [
+                {
+                  id: '1',
+                  description: 'New task description',
+                  status: 'UNCOMPLETED',
+                  title: 'New task title ',
+                },
+                {
+                  id: '2',
+                  description: 'New task description 2',
+                  status: 'COMPLETED',
+                  title: 'New task title 2',
+                },
+              ],
+              newbieTasks: [
+                {
+                  id: '3',
+                  description: 'New task description',
+                  status: 'COMPLETED',
+                  title: 'New task title ',
+                },
+                {
+                  id: '4',
+                  description: 'New task description 2',
+                  status: 'COMPLETED',
+                  title: 'New task title 2',
+                },
+              ],
+            },
+          },
+        },
+      },
+    ];
+
+    test('renders correctly', async () => {
+      const component = create(
+        <MockedProvider mocks={taskListWithDataResponse} addTypename={false}>
+          <MemoryRouter initialEntries={[path]}>
+            <Route path={'/buddy/newbies/:newbieId/tasks'}>
+              <TasksList />
+            </Route>
+          </MemoryRouter>
+        </MockedProvider>
+      );
+
+      expect(component.toJSON()).toMatchSnapshot();
+
+      await act(async () => {
+        await waitForExpect(() => {
+          expect(component.toJSON()).toMatchSnapshot();
+        });
+      });
+    });
+  });
+
+  describe('when there are not tasks', () => {
+    const taskListEmptyStateResponse = [
```

Comment:
I think that we started to keep those queries inside `__mocks__` folder in the https://github.com/BuildIt-Poland/buddy-app/blob/a4cb7d091aaf50ea907dae07d26ab4614a4b5f7d/src/__mocks__/graphql.ts#L20

2/2

Reply:
fixed https://github.com/BuildIt-Poland/buddy-app/pull/40/commits/4f2356f853516fb7efc33049cff09e1a8025e585